lib/src/publish_plugin_command.dart

```
@@ -0,0 +1,229 @@
+    argParser.addFlag(
+      _tagReleaseOption,
+      help: 'Whether or not to tag the release.',
+      defaultsTo: true,
+      negatable: true,
+    );
+    argParser.addFlag(
+      _pushTagsOption,
+      help:
+          'Whether or not tags should be pushed to a remote after creation. Ignored if tag-release is false.',
+      defaultsTo: true,
+      negatable: true,
+    );
+    argParser.addOption(
+      _remoteOption,
+      help:
+          'The name of the remote to push the tags to. Ignored if push-tags or tag-release is false.',
+      // Flutter convention is to use "upstream" for the single source of truth, and "origin" for personal forks.
+      defaultsTo: 'upstream',
+    );
+  }
+
+  static const String _packageOption = 'package';
+  static const String _tagReleaseOption = 'tag-release';
+  static const String _pushTagsOption = 'push-tags';
+  static const String _pubFlagsOption = 'pub-publish-flags';
+  static const String _remoteOption = 'remote';
+
+  // Version tags should follow <package-name>-v<semantic-version>. For example,
+  // `flutter_plugin_tools-v0.0.24`.
+  static const String _tagFormat = '%PACKAGE%-v%VERSION%';
+
+  @override
+  final String name = 'publish-plugin';
+
+  @override
+  final String description =
+      'Attempts to publish the given plugin and tag its release on Github.';
+
+  final Print _print;
+  final Stdin _stdin;
+  // The directory of the actual package that we are publishing.
+  Directory _packageDir;
+  StreamSubscription<String> _stdinSubscription;
+
+  @override
+  Future<Null> run() async {
+    checkSharding();
+    _print('Checking local repo...');
+    _packageDir = _checkPackageDir();
+    await _checkGitStatus();
+    final bool shouldPushTag = argResults[_pushTagsOption];
+    final String remote = argResults[_remoteOption];
+    String remoteUrl;
+    if (shouldPushTag) {
+      remoteUrl = await _verifyRemote(remote);
+    }
```

nits: seems can be moved in the tagging section.

---

It's kind of awkward code wise to use these booleans twice, but I wanted to fail fast here instead of first publishing to pub, tagging, and then failing to push. I think it would be annoying to have to suddenly deal with git manually and have things half done. WDYT?